tests/auxiliary/user-account/test_user_account.py

```diff
@@ -0,0 +1,172 @@
+        ))
+    chain = klass.from_genesis(chaindb, genesis_params, genesis_state)
+    return chain
+
+
+@pytest.fixture
+def unsigned_transaction():
+    return StandardTransaction(
+        chain_id=1,
+        shard_id=2,
+        target=b"\xaa" * 20,
+        start_gas=3,
+        access_list=[[b"\xaa" * 20, b"\x00"]],
+        receiver=b"\xbb" * 20,
+        value=4,
+        nonce=5,
+        gas_price=6,
+        msg_data=b"\xcc" * 123,
+    )
+
+
+@pytest.fixture
+def tx1():
+    return StandardTransaction(
+        chain_id=1,
+        shard_id=1,
+        target=USER_ACCOUNT_ADDRESS,
+        start_gas=200000,
+        access_list=[],
+        receiver=RECEIVER,
+        value=100,
+        nonce=0,
+        gas_price=0,
+        msg_data=b""
+    ).sign(USER_PRIVATE_KEY)
+
+
+def test_standard_transaction(unsigned_transaction):
+    # test initialization
+    assert unsigned_transaction.chain_id == 1
+    assert unsigned_transaction.shard_id == 2
+    assert unsigned_transaction.target == b"\xaa" * 20
+    assert unsigned_transaction.start_gas == 3
+    assert unsigned_transaction.access_list == [[b"\xaa" * 20, b"\x00"]]
+    assert unsigned_transaction.receiver == b"\xbb" * 20
+    assert unsigned_transaction.value == 4
+    assert unsigned_transaction.nonce == 5
+    assert unsigned_transaction.gas_price == 6
+    assert unsigned_transaction.msg_data == b"\xcc" * 123
+    assert unsigned_transaction.signature == (0, 0, 0)
+
+    # test getters and setters
+    unsigned_transaction.chain_id = 2
+    assert unsigned_transaction.chain_id == 2
+
+    unsigned_transaction.shard_id = 3
+    assert unsigned_transaction.shard_id == 3
+
+    unsigned_transaction.target = b"\xbb" * 20
+    assert unsigned_transaction.target == b"\xbb" * 20
+
+    unsigned_transaction.start_gas = 4
+    assert unsigned_transaction.start_gas == 4
+
+    unsigned_transaction.access_list = []
+    assert unsigned_transaction.access_list == []
+
+    unsigned_transaction.receiver = b"\xcc" * 20
+    assert unsigned_transaction.receiver == b"\xcc" * 20
+
+    unsigned_transaction.value = 5
+    assert unsigned_transaction.value == 5
+
+    unsigned_transaction.nonce = 6
+    assert unsigned_transaction.nonce == 6
+
+    unsigned_transaction.gas_price = 7
+    assert unsigned_transaction.gas_price == 7
+
+    unsigned_transaction.msg_data = b"\xdd" * 99
+    assert unsigned_transaction.msg_data == b"\xdd" * 99
+
+    unsigned_transaction.signature = (1, 2, 3)
+    assert unsigned_transaction.signature == (1, 2, 3)
+
+
+def test_signing(unsigned_transaction):
+    assert not unsigned_transaction.is_signed
+    private_key = keys.PrivateKey(b'\x22' * 32)
+    unsigned_transaction.sign(private_key)
+    assert unsigned_transaction.is_signed
+    assert unsigned_transaction.get_sender() == private_key.public_key.to_canonical_address()
+
+
+def test_valid_transfer(chain, tx1):
+    vm = chain.get_vm()
+    vm.apply_transaction(tx1)
```

Comment:
Are there any assertions that we could make here?

Reply:
Definitely. That's one of the reasons why it's still WiP (this test doesn't even pass yet)